app/contexts/LocalAuthContext.tsx

```diff
@@ -0,0 +1,109 @@
+import { Logging } from '@api'
+import * as LocalAuthentication from 'expo-local-authentication'
+import { AuthenticationType, LocalAuthenticationOptions, SecurityLevel } from 'expo-local-authentication'
+import React, { createContext, useCallback, useContext, useEffect, useState } from 'react'
+import { PrivacyLockPersistence } from '../api/wallet/privacy_lock'
+
+export interface PrivacyLockContextI {
+  // user's hardware condition, external
+  fetchHardwareStatus: () => void // not likely needed, in case user change device's security setting
+  hasHardware: boolean
+  hardwareSecurityLevel: SecurityLevel
+  supportedTypes: AuthenticationType[]
+  isDeviceProtected: boolean
```

Comment:
Reduced some of the API function, leaving hardware status getter here, they are required if UI trying to explain why privacy lock option is disabled for requirement not met situation.